lib/dogma/rule/infix_operator_padding.ex

```diff
@@ -0,0 +1,126 @@
+use Dogma.RuleBuilder
+
+defrule Dogma.Rule.InfixOperatorPadding do
+  @moduledoc """
+  A rule that ensures that all infix operators, except the range
+  operator `..`, are surrounded by spaces.
+
+  This rule is only enabled for Elixir v1.1 or greater.
+
+  Good:
+
+      foo = bar
+
+      foo - bar
+
+      foo || bar
+
+      foo |> bar
+
+      foo..bar
+
+  Bad:
+
+      foo=bar
+
+      foo-bar
+
+      foo||bar
+
+      foo|>bar
+  """
+
+  @operators [
+    :comp_op,
+    :comp_op2,
+    :dual_op,
+    :mult_op,
+    :two_op,
+    :arrow_op,
+    :rel_op,
+    :rel_op2,
+    :and_op,
+    :or_op,
+    :match_op,
+    :in_match_op,
+    :assoc_op,
+    :stab_op,
+    :pipe_op
+  ]
+
+  @subtracters [
+    :number,
+    :identifier
+  ]
+
+  @ignore_ops [
+    :-,
+    :..
+  ]
+
+  def test(_rule, script) do
+    if Version.compare("1.1.0", System.version) == :gt do
+      []
+    else
+      script.tokens
+      |> Enum.map(&normalize_token/1)
+      |> check_operators
+    end
+  end
+
+  defp normalize_token({a, {b, c, d}}), do: {a, b, c, d, nil}
+  defp normalize_token({a, {b, c, d}, e}), do: {a, b, c, d, e}
+  defp normalize_token({a, {b, c, d}, e, _}), do: {a, b, c, d, e}
+  defp normalize_token({a, {b, c, d}, e, _, _}), do: {a, b, c, d, e}
+
+  defp check_operators(tokens, acc \\ [])
+
+  defp check_operators([], acc), do: Enum.reverse(acc)
+
+  defp check_operators([
+    {token, line, _, _, _},
+    {:identifier, line, _, _, _},
+    {:mult_op, line, _, _, :/}
+    | rest], acc)
+  when token == :capture_op or token == :. do
+    check_operators(rest, acc)
+  end
+
+  defp check_operators([
+    {token1, line, _, end1, _},
+    {:dual_op, line, start2, end2, :-},
+    {token3, line, start3, _, _}
+    | rest], acc)
+  when (token1 in @subtracters or token1 == :")")
+  and (token3 in @subtracters or token3 == :"(")
+  and (end1 == start2 or end2 == start3) do
```

Comment:
You can assign these value to the same variable you don't need to do the equality check.

i.e. `def foo(a, a)` is the same as `def foo(a, b) when a == b`